store/etcd/proto.go

```diff
@@ -168,16 +163,24 @@ func valueFromProtobuf(v *pb.Value) *rule.Value {
 	}
 }
 
+// signatureToProtobuf creates a protobuf Signature from a regula.Signature.
+func signatureToProtobuf(sig *regula.Signature) *pb.Signature {
+	return &pb.Signature{
+		ReturnType: sig.ReturnType,
+		ParamTypes: sig.ParamTypes,
+	}
+}
+
 // signatureFromProtobuf creates a regula.Signature from a protobuf Signature.
 func signatureFromProtobuf(s *pb.Signature) *regula.Signature {
-	sig := &regula.Signature{
+	sig := regula.Signature{
 		ReturnType: s.ReturnType,
-		ParamTypes: make(map[string]string),
+		ParamTypes: s.ParamTypes,
 	}
 
-	for k, v := range s.ParamTypes {
-		sig.ParamTypes[k] = v
+	if sig.ParamTypes == nil {
+		sig.ParamTypes = make(map[string]string)
 	}
 
-	return sig
+	return &sig
```

Comment:
📖 What is the difference between:
```golang
sig := &regula.Signature{...}
...
return sig
```

And
```golang
sig := regula.Signature{...}
...
return &sig
```

Reply:
It's the same thing